feat (notification): hide subtype system from logged in user

MaxErikssonSK opened the pull request:
Notification with subtype system and triggered by sign in user will be hidden.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [ ] I have added/updated tests to cover my changes (if applicable).

Jira: https://jira.sundsvall.se/browse/UF-16558

MaxErikssonSK committed:
feat (notification): hide subtype system from logged in user

Notification with subtype system and triggered by sign in user will be hidden.

OliverBorgstrom commented on frontend/src/components/notifications/notifications-wrapper.tsx:
```diff
@@ -35,8 +36,15 @@ export const NotificationsWrapper: React.FC<{
+    const createdBy = (n.createdBy || '').toLowerCase();
+    const currentUser = (user?.username || '').toLowerCase();
+
+    return !(n.subType === 'SYSTEM' && (createdBy === currentUser || createdBy === 'unknown'));
```
Vet inte om det har någon betydelse, men det är uppercase i casedatanotification  "createdBy": "UNKNOWN",

MaxErikssonSK replied:
Borde inte vara ett problem för createdBy sätter vi till .toLowerCase() oavsätt vad det är.

MaxErikssonSK committed:
feat: update badge counter logic

- Moved logic to util file.
- Now the badge counter is synced with the filter logic for notifications.
